Framework/Muon/test/ApplyMuonDetectorGroupingTest.h

```diff
@@ -516,125 +208,65 @@ class ApplyMuonDetectorGroupingTest : public CxxTest::TestSuite {
-
-    std::string emptyString("");
+  void
+  test_grouping_with_multiple_detectors_and_asymmetry_analysis_gives_correct_values() {
 
-    MatrixWorkspace_sptr ws = createAsymmetryWorkspace(5, 10);
-    AnalysisDataService::Instance().addOrReplace("inputData", ws);
+    MatrixWorkspace_sptr ws = createAsymmetryWorkspace(3, 10);
+    setUpADSWithWorkspace setup(ws);
+    Mantid::API::IAlgorithm_sptr alg =
+        algorithmWithPropertiesSet("inputData", "inputGroup");
+    alg->setProperty("AnalysisType", "Asymmetry");
+    alg->setProperty("Grouping", "1,2,3");
+    alg->execute();
 
-    auto wsGroup = boost::make_shared<WorkspaceGroup>();
-    AnalysisDataService::Instance().addOrReplace("inputGroup", wsGroup);
-
-    std::string wsName = ws->getName();
-    std::string wsGroupName = (wsGroup)->getName();
-
-    TS_ASSERT_EQUALS(wsGroup->getNumberOfEntries(), 0);
-
-    ApplyMuonDetectorGrouping alg;
-    TS_ASSERT_THROWS_NOTHING(alg.initialize());
-    TS_ASSERT(alg.isInitialized());
-
-    TS_ASSERT_THROWS_NOTHING(alg.setProperty("InputWorkspace", wsName));
-    TS_ASSERT_THROWS_NOTHING(
-        alg.setProperty("InputWorkspaceGroup", wsGroupName));
-    TS_ASSERT_THROWS_NOTHING(alg.setProperty("groupName", "test2"));
-    TS_ASSERT_THROWS_NOTHING(alg.setProperty("Grouping", "1,2,3"));
-    TS_ASSERT_THROWS_NOTHING(alg.setProperty("AnalysisType", "Asymmetry"));
-    TS_ASSERT_THROWS_NOTHING(alg.setProperty("TimeMin", 0.0));
-    TS_ASSERT_THROWS_NOTHING(alg.setProperty("TimeMax", 30.0));
-    TS_ASSERT_THROWS_NOTHING(alg.setProperty("RebinArgs", emptyString));
-    TS_ASSERT_THROWS_NOTHING(alg.setProperty("TimeOffset", 0.0));
-    TS_ASSERT_THROWS_NOTHING(
-        alg.setProperty("SummedPeriods", std::to_string(1)));
-    TS_ASSERT_THROWS_NOTHING(alg.setProperty("SubtractedPeriods", emptyString));
-    TS_ASSERT_THROWS_NOTHING(alg.execute());
-    TS_ASSERT(alg.isExecuted());
-
-    TS_ASSERT(wsGroup);
-    if (wsGroup) {
-      TS_ASSERT_EQUALS(wsGroup->getNumberOfEntries(), 2);
-    }
-
-    TS_ASSERT(wsGroup->getItem("inputGroup; Group; test2; Asym; #1"));
-    TS_ASSERT(wsGroup->getItem("inputGroup; Group; test2; Asym; #1_Raw"));
+    WorkspaceGroup_sptr wsGroup = boost::dynamic_pointer_cast<WorkspaceGroup>(
+        AnalysisDataService::Instance().retrieve("inputGroup"));
 
     auto wsOut = boost::dynamic_pointer_cast<MatrixWorkspace>(
-        wsGroup->getItem("inputGroup; Group; test2; Asym; #1_Raw"));
+        wsGroup->getItem("inputGroup; Group; test; Asym; #1_Raw"));
 
     // Check values against calculation by hand.
     TS_ASSERT_DELTA(wsOut->readX(0)[0], 0.000, 0.001);
     TS_ASSERT_DELTA(wsOut->readX(0)[4], 0.400, 0.001);
     TS_ASSERT_DELTA(wsOut->readX(0)[9], 0.900, 0.001);
 
-    TS_ASSERT_DELTA(wsOut->readY(0)[0], -0.12048, 0.00001);
-    TS_ASSERT_DELTA(wsOut->readY(0)[4], -0.11211, 0.00001);
-    TS_ASSERT_DELTA(wsOut->readY(0)[9], -0.09927, 0.00001);
+    TS_ASSERT_DELTA(wsOut->readY(0)[0], 2.101, 0.001);
```

Comment:
why have these numbers changed?

Reply:
The generated 'fake' data has changed, for one thing we no longer allow negative counts.